ISSUE_TEMPLATE.md

```diff
@@ -1,36 +1,62 @@
+----------------------------- Delete below -----------------------------
 
-Please fill the info fields, it helps to get you faster support ;)
+INSTRUCTIONS
+============
+If you do not follow these instructions, your issue may be dismissed.
+
+1. Follow the checklist and fill in the [ ] spaces with an X.
+2. Fill in all the fields marked with [ ] (pick the correct option for you in each case, delete the others).
+3. If you haven't already done so, test your issue against current master branch (aka latest git), because it may have been already fixed.
+4. Describe your problem.
+5. If you have a STACK DUMP decode it:
 
-if you have a stack dump decode it:
 https://github.com/esp8266/Arduino/blob/master/doc/Troubleshooting/stack_dump.rst
 
-for better debug messages:
-https://github.com/esp8266/Arduino/blob/master/doc/Troubleshooting/debugging.rst
+6. Include a Minimal Complete Reproducible Example sketch that shows your issue. Do not include your entire project, or a huge piece of code.
+7. Include debug messages:
+
+https://github.com/esp8266/Arduino/blob/master/doc/Troubleshooting/debugging.rst 
+
+8. Use markup (buttons above) and the Preview tab to check what the issue will look like.
+9. Delete these instructions from the above to the below marker lines before submitting this issue.
+
 
------------------------------ Remove above -----------------------------
+----------------------------- Delete above -----------------------------
 
 ### Basic Infos
 
-#### Hardware
-Hardware:			?ESP-12?
-Core Version:      	?2.1.0-rc2?
+- [ ] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
+- [ ] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
+- [ ] I have tested that the issue is present in current master branch (aka latest git).

```

Comment:
That might be a bit harsh, no? I think it's fair that we also accept issue reports for release versions and ask to test with latest git if needed. How about "I have searched the issue tracker for a similar issue" instead?

Reply:
Normally I'd agree, but our release model is linear wrt to development, i.e.: our releases are a snapshot of master, and we don't use branches for released versions (the typical "stable release" model). As a result, It doesn't make much sense to track issues that apply to a particular release, except maybe the very last release, because there are no release branches to port a fix to.
Whatever else, the current reality is that we're getting a whole lot of duplicate issues. A release has an issue, we get a report, we fix it in master, then we keep getting multiple reports for the same issue, and we can't always tell that all those additional issues are the same one. Then, when an issue comes up that isn't obvious, the first thing we do is ask to test with latest git anyways. This wastes a whole lot of time, both on our end because of dealing with those issues, and on the user's end because of the turnaround and turnaround back again.

My thinking is that installing git is easy enough, and it's even documented. The user is supposed to do his homework, but many are a bit lazy, or just don't know, so don't do enough research prior to opening an issue. So anything we can do to "encourage" a user to do due diligence, that reduces our own overhead, is good from my pov.
So I think your idea to search the issue tracker is a good one, but as an additional item.